### PR TITLE
Display SUMA updates in Host Details page

### PR DESCRIPTION
**Pull request by jamie-suse**
# Description

This change mounts the `AvailableSoftwareUpdates` component in the _Host Details_ page, allowing users to see the available updates and patches for a host, as instructed by SUMA.

## How was this tested?

Added React component tests to validate that the _Available Software Updates_ component is correctly being displayed in the _Host Details_ page.

### CODE REVIEW

File: assets/js/pages/HostDetailsPage/HostDetailsPage.jsx [hunk 1]
```diff
@@ -92,6 +103,9 @@ function HostDetailsPage() {
       selectedChecks={hostSelectedChecks}
       slesSubscriptions={host.sles_subscriptions}
       catalog={catalog}
+      relevantPatches={numRelevantPatches}
+      upgradablePackages={numUpgradablePackages}
+      softwareUpdatesLoading={softwareUpdatesLoading}
```
Comment (dottorblaster) [inline assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:108]: Maybe we wanna add a tooltip text as a prop here? I really don't know, and I really don't wanna block this PR just for this nit

Comment (jamie-suse) [reply assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:110]: Done, do you think it's worth testing the tooltip behaviour from `HostDetails.test.jsx`, as it's already being tested in `AvailableSoftwareUpdates.test.jsx`... :thinking:

Comment (dottorblaster) [reply assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:110]: No, I think we can iterate over that, for example in the future we can have a selector to determine tooltip's text

File: assets/js/pages/HostDetailsPage/HostDetailsPage.jsx [hunk 2]
```diff
@@ -66,6 +78,7 @@ function HostDetailsPage() {
     getExportersStatus();
     refreshCatalog();
     dispatch(updateLastExecution(hostID));
+    dispatch(fetchSoftwareUpdates(hostID));
```
Comment (nelsonkopliku) [inline assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:81]: this needs to be like the following
```suggestion
    dispatch(fetchSoftwareUpdates({hostId: hostID}))
```
unless we change that action

Comment (jamie-suse) [reply assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:81]: Hmm just had a look through our codebase and we use a mix of `hostID` and `hostId` :thinking: I guess `hostID` is *more* popular though, so I'll change the action

Comment (nelsonkopliku) [reply assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:81]: up to you, but the thing is we need to pass an object, not the plain id (whatever is called).
unless we do transformation in the action

```
export const createAction = createAction(
  FETCH_SOFTWARE_UPDATES,
  (hostId) => ({ payload: { hostId } })
);
```

File: assets/js/pages/HostDetailsPage/HostDetails.jsx [hunk 3]
```diff
@@ -215,6 +220,13 @@ function HostDetails({
             />
           </div>
         </div>
+        <AvailableSoftwareUpdates
```
Comment (nelsonkopliku) [inline assets/js/pages/HostDetailsPage/HostDetails.jsx:223]: let's gate this behind a check against `getFromConfig('suseManagerEnabled')`

File: assets/js/pages/HostDetailsPage/HostDetailsPage.jsx (hunk 2 above)
Comment (jamie-suse) [reply assets/js/pages/HostDetailsPage/HostDetailsPage.jsx:81]: Ah I see what you mean. I thought I was using the action-creator in `@state/softwareUpdates`. Instead, I was calling the Saga directly :flushed:

**Review by dottorblaster — APPROVED** [review]

**Pull request merged**